pkg/utils/utils.go

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"os"
+	"path/filepath"
+)
+
+func CacheDir() string {
+	tmpDir, err := os.UserCacheDir()
+	if err != nil {
+		tmpDir = os.TempDir()
+	}
+	return filepath.Join(tmpDir, "tekton")
+}
+
+func ConfigDir() (string, error) {
+	homeDir, err := os.UserHomeDir()
```

Comment:
Actually, this is breaking change isn't it ? it will return `$HOME/tkn`. It should be `os.UserConfigDir` instead.

Reply:
Actually looking at the code for `UserConfigDir` it uses `~/Library` paths for Darwin. so config would be `$HOME/Library/Application` even if `XDG_CONFIG_HOME` was set...

Shall I avoid using the `os.UserXDir` functions and write out the logic myself?

Reply:
At least, `os.UserHomeDir()` on Linux will return `$HOME` and thus this function returns `$HOME/tkn` which is wrong.
For darwin, `$HOME/Library/Application` is good for me but.. I am not a user for darwin so… cc @chmouel

Reply:
I've corrected this to `UserConfigDir` but we still need to decide on how to handle Mac so I'll leave this open

Reply:
@vdemeester not a OSX user anymore as well so I can't confirm